### PR TITLE
UPDATE authentication

### PR DESCRIPTION
**Pull request by Duc-Quang-0310**

### CODE REVIEW

File: src/helper/encode.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { sign, verify } from "jsonwebtoken";
+
+const HASH_STRING = "TruongBaoNgoc";
```
Comment (hieuvmo) [inline src/helper/encode.ts:3]: -_-

File: src/pages/sign-in/SignIn.tsx [hunk 2]
```diff
@@ -51,7 +59,16 @@ const SignIn = () => {
     },
   });
 
-  const handleSubmitLogin = (body: LoginBody) => login(body);
+  const handleSubmitLogin = (body: LoginBody) => {
+    if (checkedRemember) {
+      localStorage.setItem(
+        "savedAccount",
```
Comment (hieuvmo) [inline src/pages/sign-in/SignIn.tsx:66]: You should store **savedAccount** in constants file

File: src/pages/sign-in/SignIn.tsx [hunk 3]
```diff
@@ -144,7 +171,13 @@ const SignIn = () => {
             custom={3}
             animate={controls}
           >
-            <Checkbox size="md" colorScheme="teal" disabled={loading}>
+            <Checkbox
+              size="md"
+              colorScheme="teal"
+              disabled={loading}
+              onChange={(e) => setCheckedRemember(e.target.checked)}
```
Comment (hieuvmo) [inline src/pages/sign-in/SignIn.tsx]: You should use function instead of setValue directly

File: src/pages/sign-in/SignIn.tsx (hunk 2 above)
Comment (Duc-Quang-0310) [reply src/pages/sign-in/SignIn.tsx:66]: Only call when its needed

**Pull request merged**